src/web-ui/src/public/Main.vue

```diff
@@ -1,249 +1,104 @@
-const ExperimentFeature = 'home_product_recs'
+const ProductsRepository = RepositoryFactory.get('products');
+const RecommendationsRepository = RepositoryFactory.get('recommendations');
+const MAX_RECOMMENDATIONS = 9;
+const EXPERIMENT_FEATURE = 'home_product_recs';
 
 export default {
   name: 'Main',
   components: {
-    Product,
-    Category
+    Layout,
+    RecommendedProductsSection,
   },
   data() {
     return {
-      feature: ExperimentFeature,
-      categories: [],
-      guest_recommended: [],
-      user_recommended: [],
-      errors: [],
-      display: '',
-      explain_recommended: '',
-      active_experiment: false,
-      personalized: false
-    }
+      feature: EXPERIMENT_FEATURE,
+      featuredProducts: null,
+      userRecommendations: null,
+      explainRecommended: null,
+    };
   },
-  async created () {
-    this.getRecommendations()
-    this.getCategories()
+  computed: { ...mapState(['user']), ...mapGetters(['personalizeUserID', 'personalizeRecommendationsForVisitor']) },
+  async created() {
+    this.getFeaturedProducts();
+    this.getUserRecommendations();
   },
   methods: {
-    async getRecommendations() {
-      // Baseline assumption that we're displaying featured products
-      this.display = 'featured'
-      if (this.personalizeUserID) {
-        this.getUserRecommendations()
-      }
-      else {
-        const { data } = await ProductsRepository.getFeatured()
-        this.guest_recommended = data.slice(0, MaxRecommendations)
-      }
+    async getFeaturedProducts() {
+      const { data } = await ProductsRepository.getFeatured();
+
+      this.featuredProducts = data.slice(0, MAX_RECOMMENDATIONS).map((product) => ({ product }));
     },
     async getUserRecommendations() {
-      const response = await RecommendationsRepository.getRecommendationsForUser(this.personalizeUserID, '', MaxRecommendations, ExperimentFeature)
+      if (!this.personalizeRecommendationsForVisitor) return;
+
+      const response = await RecommendationsRepository.getRecommendationsForUser(
+        this.personalizeUserID,
+        '',
+        MAX_RECOMMENDATIONS,
+        EXPERIMENT_FEATURE,
+      );
 
       if (response.headers) {
-        if (response.headers['x-personalize-recipe']) {
-          if (this.personalizeRecommendationsForVisitor) {
-            // Expect recommendations to be personalized by now
-            this.display = 'Inspired by your shopping trends';
-          }
-          else {
-            // Assume popular products are being displayed until user generates some events
-            this.display = 'Trending products';
-          }
-          this.personalized = true
-          this.explain_recommended = 'Personalize recipe: ' + response.headers['x-personalize-recipe']
-        }
-        if (response.headers['x-experiment-name']) {
-          // Can't be sure if we're personalizing or not and if so to what degree
-          this.display = 'Recommended for you';
```

Comment:
It doesn't appear that the new cold user section header behavior was ported correctly to the new UX. There are now 4 possible display headings: "Featured", "Trending products", "Inspired by your shopping trends", and "Recommended for you". Looks like the PR has us back to the old "Featured" and "Inspired by your shopping trends".

Reply:
Updated to dynamically display 1 of the 3 possible headings within `getUserRecommendations` now, in addition to the featured products section regardless of authentication.